File: crates/periskop-static-scanner/fixtures/go/negative/internal_service.go
// Looks like an egress call and is not one.
//
// The destination is an internal service. A rule that fired here would report
// every HTTP call in a Go codebase, and a report full of those trains the reader
// to skim past the findings that matter.
package negative

import (
	"net/http"
	"strings"
)

func enrich(record string) (*http.Response, error) {
	body := strings.NewReader(record)
	return http.Post("https://billing.internal.example/v1/enrich", "application/json", body)
}

File: crates/periskop-static-scanner/fixtures/python/negative/internal_service.py
"""Looks like an egress call and is not one.

The destination is an internal service. A rule that fires here would report every
HTTP call in a codebase, and a report full of those trains the reader to skim.
"""

import requests


def enrich(record):
    return requests.post("https://billing.internal.example/v1/enrich", json=record)
